tests/item.spec.js

```diff
@@ -55,13 +59,19 @@ describe('Item', function () {
       done()
     })
   })
-
+  
   describe('_generateXML', function () {
     it('should calculate netUnitPrice', function (done) {
       expect(soldItem1._options).to.have.property('netUnitPrice').that.is.a('number')
       done()
     })
 
+    it('should set netUnitPrice or grossUnitPrice to 0', function (done) {
+      expect(soldItem1ZeroNetPrice._options).to.have.property('netUnitPrice').that.is.a('number').to.equal(0)
```

Comment:
Technically, you just moved the test here. But still does not test the xml generation process.

In my mind the proper test is
``` javascript
it ('should support 0 net unit price', function (done) {
  const item = new Item({
    label: 'First item',
    quantity: 2,
    unit: 'qt',
    vat: 27, // can be a number or a special string
    netUnitPrice: 0, // calculates gross and net values from per item net
    comment: 'An item'
  })

  const xml = item._generateXML(null, invoice._options.currency)
  parser.parseString('<wrapper>' + xml + '</wrapper>', function (err, result) {
    if (err) {
      return done(err)
    }

    const tetel = result.wrapper.tetel[0]
    // use '0' string comparison because the xml parses elements as string
    expect(tetel.nettoErtek).to.deep.equal(['0'])
    expect(tetel.bruttoErtek).to.deep.equal(['0'])
    expect(tetel.afaErtek).to.deep.equal(['0'])

    done()
  })
})
```

because it test what is the result of the `_generateXML` method.
Of course for the proper testing have to validate the other result properties.

We need the 0 gross price test and not necessary the `createSoldItemNetZero` and `createSoldItemGrossZero` helpers because creating an `Item` object is not a complicated process and we don't reuse it elsewhere.